The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5466422741/jobs/9951321020) [(plain)](https://github.com/rust-lang/rust/commit/e1e04a8beed95e78b1391c7f4650fb68abd86f8f/checks/14800061478/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
    Checking addr2line v0.20.0
error[E0599]: no method named `try_destructure_mir_constant` found for struct `TyCtxt<'tcx>` in the current scope
   --> src/tools/clippy/clippy_utils/src/consts.rs:728:31
    |
728 |     if let Some(dc) = lcx.tcx.try_destructure_mir_constant(lcx.param_env.and(result))
    |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: there is a method with a similar name: `destructure_const`
    Checking dirs-sys-next v0.1.2
    Checking winapi-util v0.1.5
    Checking termize v0.1.1
    Checking atty v0.2.14
```

</details>

--- a/src/tools/clippy/clippy_utils/src/consts.rs
+++ b/src/tools/clippy/clippy_utils/src/consts.rs
@@ -725,13 +725,14 @@ fn field_of_struct<'tcx>(
     result: mir::ConstantKind<'tcx>,
     field: &Ident,
 ) -> Option<mir::ConstantKind<'tcx>> {
-    if let Some(dc) = lcx.tcx.try_destructure_mir_constant(lcx.param_env.and(result))
+    if let mir::ConstantKind::Val(result, ty) = result
+        && let Some(dc) = lcx.tcx.try_destructure_mir_constant_for_diagnostics((result, ty))
         && let Some(dc_variant) = dc.variant
         && let Some(variant) = adt_def.variants().get(dc_variant)
         && let Some(field_idx) = variant.fields.iter().position(|el| el.name == field.name)
-        && let Some(dc_field) = dc.fields.get(field_idx)
+        && let Some(&(val, ty)) = dc.fields.get(field_idx)
     {
-        Some(*dc_field)
+        Some(mir::ConstantKind::Val(val, ty))
     }
     else {
         None

